orchestra/tests/test_task_lifecycle.py

```diff
@@ -17,16 +19,19 @@
 from orchestra.tests.helpers.fixtures import setup_models
 from orchestra.tests.helpers.fixtures import StepFactory
 from orchestra.utils.assignment_snapshots import empty_snapshots
```

Comment:
high-level: i'd feel more comfortable about these new tests if they tested at the views.py/request level rather than internal apis.  while we try to not include business logic in views.py, we inevitably fail, and ensuring we test this (plus error conditions) keeps us less brittle.